eth/eth2.go

```diff
@@ -0,0 +1,325 @@
+// Copyright 2020 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package eth
+
+import (
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+	chainParams "github.com/ethereum/go-ethereum/params"
+	"github.com/ethereum/go-ethereum/trie"
+)
+
+type Eth2API struct {
+	eth  *Ethereum
+	env  *eth2bpenv
+	head common.Hash
+}
+
+// NewEth2API creates a new API definition for the eth2 prototype.
+func NewEth2API(eth *Ethereum) *Eth2API {
+	return &Eth2API{eth: eth}
+}
+
+type eth2bpenv struct {
+	state   *state.StateDB
+	tcount  int
+	gasPool *core.GasPool
+
+	header   *types.Header
+	txs      []*types.Transaction
+	receipts []*types.Receipt
+}
+
+func (api *Eth2API) commitTransaction(tx *types.Transaction, coinbase common.Address) error {
+	//snap := eth2rpc.current.state.Snapshot()
```

Comment:
Any particular reason to remove this "snapshot" mechanism?

Reply:
yes: it didn't work and wasn't needed for the prototype

Reply:
I don't think it makes sense to try activating it now, this part will be replaced with your PR soon enough.